common/quick_panel.py

```diff
@@ -81,45 +103,43 @@ def on_selection(self, index):
         if index == self.limit:
             self.skip = self.skip + self.limit
             sublime.set_timeout(self.show, 10)
+        elif self.ret_list:
+            if index == -1:
+                self.on_done(None)
+            else:
+                self.on_done(self.ret_list[index])
         else:
             self.on_done(self.skip + index)
 
 
 class LogPanel(PaginatedPanel):
 
     def __init__(self, items, on_done, limit=6000, selected_index=None):
-        self.limit = limit
-        self.items = (entry for entry in items)
-        self.on_done = on_done
-        self.limit = limit
-        self.flags = sublime.MONOSPACE_FONT | sublime.KEEP_OPEN_ON_FOCUS_LOST
-        self.selected_index = selected_index
-        self.on_highlight = self.on_entry_highlight
+        flags = sublime.MONOSPACE_FONT | sublime.KEEP_OPEN_ON_FOCUS_LOST
+        super().__init__(items, on_done, flags=flags, selected_index=selected_index, limit=limit)
```

Comment:
Another way to resolve the mentioned "too many kwargs" code-smell, in case of classes that are to be inherited from, is to move these "options" to class attributes. For example for flags we would do this:

```python
class PaginatedPanel(object):
    flags = None


class LogPanel(PaginatedPanel):
    flags = sublime.MONOSPACE_FONT | sublime.KEEP_OPEN_ON_FOCUS_LOST
```

And then the `__init__` can be removed from `LogPanel`. Afterwards, `PaginatedPanel` can be cleaned up to include parameters that are really required for generally initializing the panel. It's `__init__` method can also accept `**kwargs`, like you have initially, to allow override the options (class attributes) from these as well:

```python
class PaginatedPanel(object):
    flags = None
    # ... more options
    
    def __init__(self, items, on_done, **kwargs):
        self.items = items
        self.on_done = on_done

        for option in ['flags', 'selected_index', 'on_highlight',
        			   'limit', 'next_message', ]:
            if option in kwargs:
                setattr(self, option, kwargs[option])

```